Justin/24_Swap_Nodes_in_Pair/swapPairs.cpp

```diff
@@ -0,0 +1,37 @@
+struct ListNode {
+      int val;
+      ListNode *next;
+      ListNode() : val(0), next(nullptr) {}
+      ListNode(int x) : val(x), next(nullptr) {}
+      ListNode(int x, ListNode *next) : val(x), next(next) {}
+ };
+
+class Solution {
+public:
+    ListNode* swapPairs(ListNode* head) {
+        if(head == nullptr || head->next == nullptr) return head;
+        ListNode *node1 = head;
+        ListNode *node2 = head->next;
+        ListNode *node3 = node2->next;
+        ListNode *ans = node2;
+        node2->next = node1;
+        node1->next = node3;
+        head = node3;
+        ListNode *preNode = node1;
+        while(head!=nullptr){
+            node1 = head;
+            node2 = head->next;
+            if(node2!=nullptr){
+                ListNode *node3 = node2->next;
```

Comment:
```suggestion
                ListNode *node3 = node2->next;
```
這邊看起來 node3 是重複宣告了？
不過是不影響結果

另外 head 跟 node3 看起來都是儲存 swap pairs 的下一個 node，應該是可以不用另外宣告 node3 改用 head 暫存就好

Reply:
對欸，重複宣告了 🙈
也可以用 head 存，我是覺得可讀性比較好？ 但是的確多了一個記憶體位置在記ＸＤ